src/main/java/duke/domain/Task.java

```diff
@@ -0,0 +1,179 @@
+package duke.domain;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.Objects;
+
+import duke.exceptions.InvalidDateTimeException;
+import duke.exceptions.InvalidTaskSpecificationException;
+
+/**
+ * The type Task.
+ */
+public class Task {
+
+    /**
+     * The constant FORMATTER.
+     */
+    protected static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern(
+            "dd-MM-yyyy HH:mm");
+    protected static final String NULL_SYMBOL = "$_$";
+    private final String text;
+    private Boolean isComplete;
+
+    /**
+     * Instantiates a new Task.
+     *
+     * @param text
+     *            the text
+     */
+    protected Task(String text) {
+        this.text = text;
+        this.isComplete = false;
+    }
+
+    /**
+     * Sets the task to be complete.
+     */
+    public void setComplete() {
+        this.isComplete = true;
+    }
+
+    /**
+     * Sets the task to be incomplete.
+     */
+    public void setIncomplete() {
+        this.isComplete = false;
+    }
+
+    @Override
+    public String toString() {
+        return this.isComplete
+                ? String.format("[X] %s", this.text)
+                : String.format("[ ] %s", this.text);
+    }
+
+    private static LocalDateTime interpretLocalDateTime(String input)
+            throws DateTimeParseException {
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern(
+                "dd-MM-yyyy HH:mm");
+        return LocalDateTime.parse(input, formatter);
+    }
+
+    /**
+     * Of task.
+     *
+     * @param type
+     *            the type
+     * @param done
+     *            the done
+     * @param text
+     *            the text
+     * @param dateTimeInput
+     *            the date time input
+     * @return the task
+     * @throws InvalidDateTimeException
+     *             the invalid date time exception
+     * @throws InvalidTaskSpecificationException
+     *             the invalid task specification exception
+     */
+    public static Task of(
```

Comment:
Some of the indentations in the first part of this method; perhaps you might want to change some of it to improve the readability of the code? Otherwise, it may be mistaken as a form of "Deep Nesting".

Reply:
I see I will see what I can do